src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -230,30 +236,109 @@ private static TwoWindingsTransformerMapData toMapData(TwoWindingsTransformer tr
+                .voltageRegulation(tapChanger.isRegulating())
+                .loadTapChangingCapabilities(tapChanger.hasLoadTapChangingCapabilities())
+                .regulatingTerminalConnectableId(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getConnectable().getId() : null)
+                .regulatingTerminalConnectableType(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getConnectable().getType().name() : null)
+                .regulatingTerminalVlId(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getVoltageLevel().getId() : null)
+                .ratioTapSteps(toMapDataRatioStep(tapChanger.getAllSteps()));
+
+        if (!Double.isNaN(tapChanger.getTargetV())) {
+            builder.targetVoltage(tapChanger.getTargetV());
         }
+
+        if (!Double.isNaN(tapChanger.getTargetDeadband())) {
+            builder.targetDeadBand(tapChanger.getTargetDeadband());
+        }
+
         return builder.build();
     }
 
-    private static TapChangerData toMapData(TapChanger<?, ?> tapChanger) {
-        return tapChanger == null ? null : TapChangerData.builder()
-            .lowTap(tapChanger.getLowTapPosition())
-            .highTap(tapChanger.getHighTapPosition())
-            .build();
+    private static TapChangerData toMapData(PhaseTapChanger tapChanger) {
+        if (tapChanger == null) {
+            return null;
+        }
+
+        TapChangerData.TapChangerDataBuilder builder = TapChangerData.builder()
+                .lowTap(tapChanger.getLowTapPosition())
+                .highTap(tapChanger.getHighTapPosition())
+                .tapPosition(tapChanger.getTapPosition())
+                .voltageRegulation(tapChanger.isRegulating())
+                .regulationMode(tapChanger.getRegulationMode())
+                .regulatingValue(tapChanger.getRegulationValue())
+                .targetDeadBand(tapChanger.getTargetDeadband())
+                .regulatingTerminalConnectableId(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getConnectable().getId() : null)
+                .regulatingTerminalConnectableType(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getConnectable().getType().name() : null)
+                .regulatingTerminalVlId(tapChanger.getRegulationTerminal() != null ? tapChanger.getRegulationTerminal().getVoltageLevel().getId() : null)
+                .phaseTapSteps(toMapDataPhaseStep(tapChanger.getAllSteps()));
+
+        if (!Double.isNaN(tapChanger.getRegulationValue())) {
+            builder.targetVoltage(tapChanger.getRegulationValue());
+        }
+
+        if (!Double.isNaN(tapChanger.getTargetDeadband())) {
+            builder.targetDeadBand(tapChanger.getTargetDeadband());
+        }
+        return builder.build();
+    }
+
+    private static List<TapChangerStepData> toMapDataPhaseStep(Map<Integer, PhaseTapChangerStep> tapChangerStep) {
```

Comment:
since refactoring there is no more `PhaseTapChangerStep` and `RatioTapChangerStep` specialization, maybe we could apply those changes here too.

Reply:
Ok I see it comes from Powsybl... arg